### PR TITLE
Add MeterIdPrefixFunction.andThen to customize MeterIdPrefix using RequestLog

### PR DESCRIPTION
**Pull request by okue**
Preparation for https://github.com/line/armeria/issues/2762.

Motivations:
Currently, `MeterIdPrefixFunction` can be customized with [`andThen`](https://github.com/line/armeria/blob/armeria-0.99.7/core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunction.java#L179-L193) which takes a function of type `(MeterRegistry, MeterIdPrefix) -> MeterIdPrefix`.
But, without taking `RequestLog`, we cannot append header values to `MeterIdPrefix`.

Modifications:
- Add `andThen: (MeterRegistry, RequestOnlyLog, MeterIdPrefix) -> MeterIdPrefix`
- Deprecate the original version `andThen(BiFunction<MeterRegistry, MeterIdPrefix, MeterIdPrefix>)`

Results:
Users can customize `MeterIdPrefixFunction` easily like

```java
MeterIdPrefixFunction
        .ofDefault("hoge")
        .andThen((registry, log, id) -> {
            if (log instanceof RequestLog) {
                return id.withTags("grpc-status", log.responseHeaders().get("grpc-status"));
            } else {
                return id;
            }
        });
```

### CODE REVIEW

**Comment by okue** [comment]
I think it may be not bad to deprecate `andThen(BiFunction<MeterRegistry, MeterIdPrefix, MeterIdPrefix>)`.

File: core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunctionCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *  under the License.
+ */
+package com.linecorp.armeria.common.metric;
+
+import com.linecorp.armeria.common.logging.RequestOnlyLog;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Returns a {@link MeterIdPrefix}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunctionCustomizer.java:23]: How about:

`Transforms a `{@link MeterIdPrefix}` into another.`

File: core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunctionCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @see MeterIdPrefixFunction
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunctionCustomizer.java:25]: `MeterIdPrefixFunction#andThen(MeterIdPrefixFunctionCustomizer)`?

**Comment by okue** [comment]
May I annotate the original one with `@Deprecated`?

**Comment by trustin** [comment]
> May I annotate the original one with `@Deprecated`?

Yes, please. :bow:

**Comment by ikhoon** [comment]
> > May I annotate the original one with `@Deprecated`?
> 
> Yes, please. 🙇

@okue Could you delegate the deprecated method to the new one?
```java
@Deprecated
default MeterIdPrefixFunction andThen(BiFunction<MeterRegistry, MeterIdPrefix, MeterIdPrefix> function) {
    requireNonNull(function, "function");
    return andThen((registry, log, meterIdPrefix) -> function.apply(registry, meterIdPrefix));
}
```

**Pull request merged**